src/datasets/packaged_modules/json/json.py

```diff
@@ -128,6 +128,13 @@ def _generate_tables(self, files):
                                     pa_table = paj.read_json(
                                         io.BytesIO(batch), read_options=paj.ReadOptions(block_size=block_size)
                                     )
+                                    feature_columns = set(self.config.features.arrow_schema.names)
```

Comment:
`self.config.features` is `None` by default - because before reading the data we don't know which columns it contains.

Reply:
ah this was made to work when features is passed in by the user in the function. maybe i can make this dynamic so that it works even when not explicitly provided.